src/components/SelectionListWithModal/index.tsx

```diff
@@ -26,22 +27,41 @@ function SelectionListWithModal<TItem extends ListItem>(
     // We need to use isSmallScreenWidth instead of shouldUseNarrowLayout here because there is a race condition that causes shouldUseNarrowLayout to change indefinitely in this component
     // See https://github.com/Expensify/App/issues/48675 for more details
     const {isSmallScreenWidth} = useResponsiveLayout();
-    const {selectionMode} = useMobileSelectionMode(true);
+    const {selectionMode} = useMobileSelectionMode(shouldAutoTurnOff);
+    const wasSelectionOnRef = useRef(false);
+    const selectionRef = useRef(0);
```

Comment:
Can you add some comments on what are these supposed to be for and comments in the `useEffect` where you use them? their presence here makes the code more complex and their purpose is not obvious.